02_oop/src/filesysapi/fileUnixSysApi.hpp

```diff
@@ -0,0 +1,69 @@
+#ifndef FILEUNIXSYSAPI_HPP
+#define FILEUNIXSYSAPI_HPP
+
+#include <string>
+
+#include <stdio.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <dirent.h>
+
+namespace filesysapi {
+
+	std::string absolutePath(const std::string& path) {
+		static char absolutePath[PATH_MAX];
+		char *pPath = realpath(path.c_str(), absolutePath);	
+		if (pPath == absolutePath)
+			return std::string(absolutePath);
+		else
+			return std::string();	
+        }
+
+	class FileHandler {
+		DIR *mDirHandle;
+		struct dirent *mCurrentFile;
+        private:
+                void firstFile(const std::string &path) {
+			if(path.empty()) {
+				throw std::invalid_argument("Path is empty");
+				mDirHandle = nullptr;	
+			} 
+			mDirHandle = opendir(path.c_str());	
+			if (!isValid())
+				throw std::invalid_argument("Invalid in ctor filesys: " + path);	
+                }
+                void close() { if (isValid()) closedir(mDirHandle); mDirHandle = nullptr; }
```

Comment:
Specify which part of the implementation is under condition using braces or indentation (like in line 33). It will make code more readable and prevent from mistakes.